lib/pdf/collate.rb

```diff
@@ -44,13 +47,13 @@ def self.merge_pdfs(output_file, *filenames)
     temp_file_handles = []
 
     filenames.map! do |filename|
-      next filename unless URI.parse(filename).scheme == 'http'
+      next filename unless string_is_url(filename)
       begin
         local_file = Tempfile.from_url(filename)
       rescue Exception => msg
         puts "Error downloading PDF file #{filename} for output file #{output_file}. Aborting"
         puts "Error message: #{msg}"
-        exit 1
+        raise
```

Comment:
backpedaling again -- I see you ARE catching / logging in collate_pdfs.rake. This looks fine then :)